woocommerce/woocommerce-ios
# Variations: Track events

Part of #3703 

# Why

Since the variations feature is almost(few tasks missing) done, now it's time to add tracking to events defined in #3703.

# How 
- Define event ids in `WooAnalyticStats`
- Define events method in `WooAnalyticsEvent`
- Update `ViewControllers` and `ViewModels` to add the tracking calls.

Note: In `ProductFormViewController` I used `ServiceLocator.analytics` instead of injecting it as a dependency, because the rest of the view controller already uses `ServiceLocator.analytics` and I think it's best to keep consistency and refactor them in one go eventually.

# Screenshot
<img width="1915" alt="Screen Shot 2021-02-25 at 2 54 27 PM" src="https://user-images.githubusercontent.com/562080/109210906-7dc9ec00-777b-11eb-8ed7-54bdb29f0d0b.png">

# Testing Steps

Go through the add and edit variation flow and see the events being tracked in the console for the following actions:
- Tap Add Variation button when there aren't variations
- Create one or more attributes
- Create attribute success or fail
- Create variation
- Create variation success or fail
- Create more variations button tapped
- Edit attributes button tapped
- Add attribute button tapped
- Delete Attribute button tapped
- Update variations attribute options row tapped
- Update variations attribute options done button tapped
- Remove product variation button tapped

Update release notes:
